### PR TITLE
Cluster password policy and user management support

### PR DESCRIPTION
**Pull request by jamesreprise**
## What is the goal of this PR?

Add functionality relevant to the changes made in https://github.com/vaticle/typedb-cluster/pull/456:
* Optional password expiry in days for each user.
* The ability for admins to set passwords of other users
* User password update, which requires old and new password

## What are the changes implemented in this PR?

We've added the functionality that is being added in a recent TypeDB Cluster pull request:
* Support updating password now requiring both the old and the new password.
* Add support for retrieving the user password expiry in days. This is optional, depending on whether password expiration is enabled on the server.
* Allow updating password as the admin requiring only a new password.

### CODE REVIEW

File: common/rpc/RequestBuilder.java
Comment (flyingsilverfin) [inline common/rpc/RequestBuilder.java:157]: if you notice our style is to inline things that fit in one line without overflow

File: connection/cluster/ClusterUserManager.java
Comment (flyingsilverfin) [inline connection/cluster/ClusterUserManager.java:98]: I recon we sohuld add a `User.of(Proto)` method to hide this `if` case away into the right domain

File: connection/cluster/ClusterUserManager.java
Comment (flyingsilverfin) [inline connection/cluster/ClusterUserManager.java:125]: Ok at this point - notice we have repeated code, then we realise there is a better domain to put the construction of a `ClusterUser` - the `ClusterUser` itself!

File: test/behaviour/connection/user/UserSteps.java
Comment (flyingsilverfin) [inline test/behaviour/connection/user/UserSteps.java:73]: what about one for `users_password_set`?

File: common/rpc/RequestBuilder.java
Comment (jamesreprise) [reply common/rpc/RequestBuilder.java:157]: This does overflow all on one line unfortunately. Something like this?
```
return ClusterUserProto.ClusterUserManager.PasswordSet.Req.newBuilder()
                        .setUsername(username).setPassword(password).build();
```

File: connection/cluster/ClusterUserManager.java
Comment (jamesreprise) [reply connection/cluster/ClusterUserManager.java:98]: Agreed, implemented.

File: connection/cluster/ClusterUserManager.java
Comment (jamesreprise) [reply connection/cluster/ClusterUserManager.java:125]: Agreed, implemented.

File: test/behaviour/connection/user/UserSteps.java
Comment (jamesreprise) [reply test/behaviour/connection/user/UserSteps.java:73]: Haven't done the behaviour tests yet, apologies.

Comment (jamesreprise) [reply test/behaviour/connection/user/UserSteps.java]: It seems as though the user behaviour steps aren't run in the current set of automation jobs in vaticle/typedb-client-java - we should decide what we want to do with these.

File: api/user/UserManager.java
Comment (flyingsilverfin) [inline api/user/UserManager.java:41]: it's pop this back up to the top, or at least near `all`

File: connection/cluster/ClusterUserManager.java
Comment (flyingsilverfin) [inline connection/cluster/ClusterUserManager.java:88]: I'd probably put a line break before one of the `.` instead of the inside the `map()` backets

**Review by flyingsilverfin — COMMENTED** [review]
Left 2 minor comments you can address - approving now conditional on those being changed.

**Review by flyingsilverfin — APPROVED** [review]

**Comment by jamesreprise** [comment]
> Left 2 minor comments you can address - approving now conditional on those being changed.

Done! Quantity of commits was due to rebasing on master.

**Pull request merged**
